common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -361,6 +366,91 @@ public static long getLong(byte[] data, int index) {
         return PlatformDependent0.getLong(data, index);
     }
 
+    private static long getLongSafe(byte[] bytes, int offset) {
+        if (BIG_ENDIAN_NATIVE_ORDER) {
+            return (long) bytes[offset] << 56 |
+                    ((long) bytes[offset + 1] & 0xff) << 48 |
+                    ((long) bytes[offset + 2] & 0xff) << 40 |
+                    ((long) bytes[offset + 3] & 0xff) << 32 |
+                    ((long) bytes[offset + 4] & 0xff) << 24 |
+                    ((long) bytes[offset + 5] & 0xff) << 16 |
+                    ((long) bytes[offset + 6] & 0xff) <<  8 |
+                    (long) bytes[offset + 7] & 0xff;
+        }
+        return (long) bytes[offset] & 0xff |
+                ((long) bytes[offset + 1] & 0xff) << 8 |
+                ((long) bytes[offset + 2] & 0xff) << 16 |
+                ((long) bytes[offset + 3] & 0xff) << 24 |
+                ((long) bytes[offset + 4] & 0xff) << 32 |
+                ((long) bytes[offset + 5] & 0xff) << 40 |
+                ((long) bytes[offset + 6] & 0xff) << 48 |
+                ((long) bytes[offset + 7] & 0xff) << 56;
+    }
+
+    private static long getLongFromBytesSafe(CharSequence bytes, int offset) {
+        if (BIG_ENDIAN_NATIVE_ORDER) {
+            // Swap the bytes within each char, and don't invert the way the array is indexed.
```

Comment:
@normanmaurer - Can you double check the byte swapping for the 3 `get*FromBytesSafe` methods here (they all take `CharSequence` parameters)?